src/etcd/__init__.py

```diff
@@ -62,13 +62,14 @@ def get_subtree(self, leaves_only=False):
             #if the current result is a leaf, return itself
             yield self
             return
-        for n in self._children:
-            node = EtcdResult(None, n)
+        else:
```

Comment:
I find it more readable to have the `else:` rather than have to pick up on the `return` in the `if:` block.